python/tests/backends/test_resource_counter.py

```diff
@@ -0,0 +1,49 @@
+# ============================================================================ #
+# Copyright (c) 2025 NVIDIA Corporation & Affiliates.                          #
+# All rights reserved.                                                         #
+#                                                                              #
+# This source code and the accompanying materials are made available under     #
+# the terms of the Apache License 2.0 which accompanies this distribution.     #
+# ============================================================================ #
+
+# Compile and run with:
+
+import cudaq
+
+
+def test_resource_counter():
+
+    @cudaq.kernel
+    def mykernel():
+        q = cudaq.qubit()
+        p = cudaq.qubit()
+
+        # Alias them
+        h(q)
+
+        m1 = mz(q)
+        if m1:
+            x(p)
+            m2 = mz(p)
+        else:
+            m3 = mz(p)
+
+    def choice():
+        return True
+
+    cudaq.set_target("quantinuum", emulate=True)
+
+    counts1 = cudaq.sample(mykernel, shots_count=5)
+    counts2 = cudaq.count_resources(choice, mykernel)
```

Comment:
Just a suggestion: looking at this example, I wonder whether we should make the `choice` callback a `kw_only` argument.
For kernels that don't have measure gates, we don't need this argument. We could use a default one that always returns `True` (and document it) :)